src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/BookingServiceTest.kt

```diff
@@ -4302,6 +4328,486 @@ class BookingServiceTest {
+          match {
+            it.crn == crn &&
+              it.premises == premises &&
+              it.arrivalDate == arrivalDate &&
+              it.departureDate == departureDate &&
+              it.application == application
+          },
+        )
+      }
+      verify(exactly = 1) {
+        mockAssessmentService.closeAssessment(user, application.id)
+      }
+    }
+
+    @Test
+    fun `createTemporaryAccommodationBooking not closing assessments and returns FieldValidationError if Application is provided and does not exist`() {
+      val crn = "CRN123"
+      val arrivalDate = LocalDate.parse("2023-02-23")
+      val departureDate = LocalDate.parse("2023-02-24")
+      val assessmentId = UUID.randomUUID()
+
+      val user = UserEntityFactory()
+        .withUnitTestControlProbationRegion()
+        .produce()
+
+      val premises = TemporaryAccommodationPremisesEntityFactory()
+        .withUnitTestControlTestProbationAreaAndLocalAuthority()
+        .produce()
+
+      val room = RoomEntityFactory()
+        .withPremises(premises)
+        .produce()
+
+      val bed = BedEntityFactory()
+        .withRoom(room)
+        .produce()
+
+      every { mockBedRepository.findByIdOrNull(bed.id) } returns bed
+
+      every { mockBookingRepository.save(any()) } answers { it.invocation.args[0] as BookingEntity }
+
+      every { mockBookingRepository.findByBedIdAndArrivingBeforeDate(bed.id, departureDate, null) } returns listOf()
+      every {
+        mockLostBedsRepository.findByBedIdAndOverlappingDate(
+          bed.id,
+          arrivalDate,
+          departureDate,
+          null,
+        )
+      } returns listOf()
+      every { mockAssessmentRepository.findByIdOrNull(assessmentId) } returns null
+      every { mockTurnaroundRepository.save(any()) } answers { it.invocation.args[0] as TurnaroundEntity }
+
+      every { mockWorkingDayCountService.addWorkingDays(any(), any()) } answers { it.invocation.args[0] as LocalDate }
+
+      val authorisableResult = bookingService.createTemporaryAccommodationBooking(
+        user,
+        premises,
+        crn,
+        "NOMS123",
+        arrivalDate,
+        departureDate,
+        bed.id,
+        assessmentId,
+        false,
+      )
+      assertThat(authorisableResult is AuthorisableActionResult.Success).isTrue
+
+      val validatableResult = (authorisableResult as AuthorisableActionResult.Success).entity
+      assertThat(validatableResult is ValidatableActionResult.FieldValidationError)
+
+      assertThat((validatableResult as ValidatableActionResult.FieldValidationError).validationMessages).contains(
+        entry("$.assessmentId", "doesNotExist"),
+      )
+
+      verify(exactly = 0) {
+        mockAssessmentService.closeAssessment(user, assessmentId)
+      }
+    }
+
+    @Test
+    fun `createTemporaryAccommodationBooking fail to save booking and not closing assessment when saving booking triggered Database exception`() {
```

Comment:
is there anyway to reduce duplication in these test methods (e.g. move the app/assessment creation into a common function?)